src/box/lua/space.cc

```diff
@@ -702,7 +702,18 @@ box_lua_space_new_or_delete(struct trigger *trigger, void *event)
 	return 0;
 }
 
+static int
+box_lua_space_free(struct trigger *trigger, void *event)
+{
+	struct lua_State *L = (struct lua_State *)trigger->data;
+	struct space *space = (struct space *)event;
+
+	luaL_unref(L, LUA_REGISTRYINDEX, space->lua_ref);
+	return 0;
+}
+
 static TRIGGER(on_alter_space_in_lua, box_lua_space_new_or_delete);
+static TRIGGER(on_space_cache_free_in_lua, box_lua_space_free);
```

Comment:
I don't see any point in avoiding this dependency. Both `box/space.c` and `box/lua/space.cc` are a part of the box lib which depends on Lua anyway.

Reply:
Dropped this commit.